bootstrap/util/configowner.go

```diff
@@ -82,8 +83,8 @@ func GetConfigOwner(ctx context.Context, c client.Client, obj metav1.Object) (*C
 			return nil, errors.Wrapf(err, "failed to parse GroupVersion from %q", ref.APIVersion)
 		}
 
-		if (ref.Kind == "Machine" || ref.Kind == "MachinePool") &&
-			refgvk.Group == clusterv1.GroupVersion.Group {
+		if (ref.Kind == "Machine" && refgvk.Group == clusterv1.GroupVersion.Group) ||
+			(ref.Kind == "MachinePool" && refgvk.Group == expv1.GroupVersion.Group) {
```

Comment:
Should there be a way to tell GetConfigOwner that a feature flag is enabled or not that allows the comparison with the experimental types?

Reply:
Yes, Vince & I were just chatting about this 😄 (same applies to everywhere MachinePool is used)

Reply:
Done in latest revision!